client/my-sites/media/details.jsx

```diff
@@ -0,0 +1,247 @@
+import VideoEditor from 'blocks/video-editor';
+import MediaActions from 'lib/media/actions';
+
+import MediaLibrarySelectedStore from 'lib/media/library-selected-store';
+import accept from 'lib/accept';
+
+class MediaDetails extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+		selected: PropTypes.number,
+		items: PropTypes.array,
+		view: PropTypes.string,
+	};
+
+	closeDetailsModal = () => {
+		page( '/media/' + this.props.site.slug + '/' );
+		return;
+	};
+
+	onEditorCancel = () => {
+		page( '/media/' + this.props.site.slug + '/' + this.getSelectedMedia().ID );
+		return;
+	};
+
+	onImageEditorDone = ( error, blob, imageEditorProps ) => {
+		if ( error ) {
+			this.onEditImageCancel( imageEditorProps );
+			return;
+		}
+
+		const { fileName, site, ID, resetAllImageEditorState } = imageEditorProps;
+
+		const mimeType = getMimeType( fileName );
+
+		const item = {
+			ID: ID,
+			media: {
+				fileName: fileName,
+				fileContents: blob,
+				mimeType: mimeType,
+			},
+		};
+
+		MediaActions.update( site.ID, item, true );
+		resetAllImageEditorState();
+		this.closeDetailsModal();
+	};
+
+	getModalButtons() {
+		// do not render buttons if the media image or video editor is opened
+		if ( this.props.view === 'edit' ) {
+			return null;
+		}
+
+		const { translate } = this.props;
+
+		return [
+			{
+				action: 'delete',
+				additionalClassNames: 'media__modal-delete-item-button is-link',
+				label: translate( 'Delete' ),
+				isPrimary: false,
+				disabled: false,
+				onClick: this.deleteMediaByItemDetail,
+			},
+			{
+				action: 'confirm',
+				label: translate( 'Done' ),
+				isPrimary: true,
+				disabled: false,
+				onClose: this.closeDetailsModal,
+			},
+		];
+	}
+
+	onVideoEditorUpdatePoster = ( { ID, posterUrl } ) => {
+		const site = this.props.site;
+
+		// Photon does not support URLs with a querystring component.
+		const urlBeforeQuery = ( posterUrl || '' ).split( '?' )[ 0 ];
+
+		if ( site ) {
+			MediaActions.edit( site.ID, {
+				ID,
+				thumbnails: {
+					fmt_hd: urlBeforeQuery,
+					fmt_dvd: urlBeforeQuery,
+					fmt_std: urlBeforeQuery,
+				},
+			} );
+		}
+
+		this.props.onBack();
+	};
+
+	restoreOriginalMedia = ( siteId, item ) => {
+		if ( ! siteId || ! item ) {
+			return;
+		}
+		MediaActions.update( siteId, { ID: item.ID, media_url: item.guid }, true );
+		this.props.onBack();
+	};
+
+	setDetailSelectedIndex = index => {
+		page( '/media/' + this.props.site.slug + '/' + this.getSelectedMedia( index ).ID );
+	};
+
+	/**
+	 * Start the process to delete media items.
+	 * `callback` is an optional parameter which will execute once the confirm dialog is accepted.
+	 * It's used especially when the item is attempting to be removed using the item detail dialog.
+	 *
+	 * @param  {Function} [callback] - callback function
+	 */
+	deleteMedia( callback ) {
+		const { translate, items } = this.props;
+		const selectedCount = items.length;
+		const confirmMessage = translate(
```

Comment:
Some bits, like this one, already appear duplicated in several places. I'm hesitant to introduce another instance of duplication; I'd rather suggest moving them to separate components and reusing them to avoid duplication.